docs/ruleset_functions/get_energy_required_to_heat_swh_use.md

```diff
@@ -0,0 +1,71 @@
+- convert the swh_water_used from whatever units it's currently in to volume (assumed to be gallons) OR, if the use units are one of the power types, calculate the energy required directly
+- if the use_units is POWER_PER_PERSON: `if use_units == "POWER_PER_PERSON":`
+  - set power equal to the swh_use_value * sum of the occupants in all of the spaces: `power = swh_use_value * sum(space.number_of_occupants for space in spaces)`
+  - in this case, swh_use_value is assumed to have units of btu/hr/person, so calculate energy_required by multiplying power (btu/hr) by the sum of the hourly_schedule: `energy_required = power * sum(hourly_schedule)`
+  - return energy_required: `return energy_required`
+- otherwise if use_units is POWER_PER_AREA: `elsif use_units == "POWER_PER_AREA":`
+  - set power equal to the swh_use_value * the sum of the floor area in all of the spaces: `power = swh_use_value * sum(space.floor_area for space in spaces)`
+  - in this case, swh_use_value is assumed to have units of btu/hr/ft2, so calculate energy_required by multiplying power (btu/hr) by the sum of the hourly_schedule: `energy_required = power * sum(hourly_schedule)`
+  - return energy_required: `return energy_required`
+- otherwise if use_units is POWER: `elsif use_units == "POWER":`
+  - in this case, swh_use_value is assumed to have units of btu/hr, so calculate energy_required by multiplying swh_use_value by the sum of the hourly_schedule: `energy_required = swh_use_value * sum(hourly_schedule)`
+  - return energy_required: `return energy_required`
+- otherwise if use_units is VOLUME_PER_PERSON: `elsif use_units == "VOLUME_PER_PERSON":`
+  - in this case swh_use_value is assumed to have the units of gallons/hr/person:
+  - set volume_flow_rate equal to swh_use_value * the sum of the occupants in all of the spaces: `volume_flow_rate = swh_use_value * sum(space.number_of_occupants for space in spaces)`
+- otherwise if use_units is VOLUME_PER_AREA: `elsif use_units == "VOLUME_PER_AREA":`
+  - in this case swh_use_value is assumed to have the units of gallons/hr/ft2:
+  - set volume_flow_rate equal to swh_use_value * the sum of the floor area in all of the spaces: `volume_flow_rate = swh_use_value * sum(space.floor_area for space in spaces)`
+- otherwise if use_units is VOLUME: `elsif use_units == "VOLUME":`
+  - in this case swh_use_value is assumed to have the units gallons/hr
+  - set volume_flow_rate equal to swh_use_value: `volume_flow_rate = swh_use_value`
+
+- set energy_required to 0: `energy_required = 0`
+- iterate through each hourly value of the hourly_schedule: `for index, hourly_value in enumerate(hourly_schedule):`
+  - set volume_this_hour equal to volume_flow_rate * hourly_value: `volume_this_hour = volume_flow_rate * hourly_value`
+  - reduce volume_this hour by the recovery efficiency: `volume_this_hour = volume_this_hour * (1-drain_heat_recovery_efficiency)`
```

Comment:
Should this also be applied above to the other units as well?

Reply:
this is applied regardless of the use units.  volume_flow_rate is calculated for each use_unit type and then this calculation is done.